smsconnection/src/test/java/com/gruppo3/smsconnection/utils/KademliaIDTest.java

```diff
@@ -0,0 +1,137 @@
+package com.gruppo3.smsconnection.utils;
+
+import android.util.Base64;
+import com.gruppo3.smsconnection.connection.exception.InvalidPeerException;
+import com.gruppo3.smsconnection.smsdatalink.message.SMSPeer;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.math.BigInteger;
+
+import static org.mockito.Mockito.*;
+
+public class KademliaIDTest {
+
+    private static String addressOne = "11223344";
+    private static String addressTwo = "12345678";
+    private static String expectedHashedAddressOne = "T58QswTP6bKxH8sTh/aU4Y8I6jU=";
+    private static byte[] expectedBytesAddressOne = new byte[]{79,-97,16,-77,4,-49,-23,-78,-79,31,-53,19,-121,-10,-108,-31,-113,8,-22,53,-116,126,-97,86,116,52,-45,-83,108,-67,127,-60};
+    private SMSPeer peerOne;
+    private SMSPeer peerTwo;
+    private Base64 mockBase64;
+    private final static int VALID_LENGTH = 20;
+    private final static int ANOTHER_LENGTH = 10;
+    private final static int NOT_VALID_LENGTH = -10;
+    private final static String expectedXorDistance = "918577119906024823139507659158657059931202283177";
```

Comment:
To be coherent with the naming convention, `final` variables should go all uppercase with underscores.
```suggestion
    private final static String EXPECTED_XOR_DISTANCE = "918577119906024823139507659158657059931202283177";
```